src/components/CheckForm/FormLayout/FormLayout.tsx

```diff
@@ -1,42 +1,117 @@
-import React, { Children, isValidElement, ReactNode } from 'react';
+import React, { Children, isValidElement, ReactNode, useState } from 'react';
 import { FieldErrors, FieldPath, useFormContext } from 'react-hook-form';
 import { GrafanaTheme2 } from '@grafana/data';
-import { useStyles2, useTheme2 } from '@grafana/ui';
+import { Button, useStyles2 } from '@grafana/ui';
 import { css, cx } from '@emotion/css';
 import { flatten } from 'flat';
 
-import { CheckFormValues } from 'types';
-import { Collapse } from 'components/Collapse';
-import { CollapseLabel } from 'components/CollapseLabel';
+import { CheckFormTypeLayoutProps, CheckFormValues } from 'types';
+
+import { FormSidebar } from './FormSidebar';
 
 type FormLayoutProps = {
   children: ReactNode;
 };
 
-export const FormLayout = ({ children }: FormLayoutProps) => {
+export const FormLayout = ({ children, formActions }: FormLayoutProps & CheckFormTypeLayoutProps) => {
   let index = -1;
+  const [activeIndex, setActiveIndex] = useState(0);
+  const sectionHeaders: Array<{ label: string; hasErrors: boolean; required: boolean }> = [];
+  const { formState, trigger } = useFormContext<CheckFormValues>();
+  const styles = useStyles2(getStyles);
 
-  return Children.map(children, (child) => {
+  let sectionCount = 0;
+  Children.forEach(children, (child) => {
+    if (!isValidElement(child)) {
+      return;
+    }
+    if (child?.type === FormSection) {
+      sectionCount++;
+    }
+  });
+  const sections = Children.map(children, (child) => {
     if (!isValidElement(child)) {
       return null;
     }
 
     if (child.type === FormSection) {
       index++;
 
-      return <FormSectionInternal {...child.props} index={index} />;
+      const errors = checkForErrors(formState.errors, child.props.fields);
+      sectionHeaders.push({ label: child.props.label, hasErrors: errors.length > 0, required: child.props.required });
+      return (
+        <FormSectionInternal
+          {...child.props}
+          index={index}
+          active={activeIndex === index}
+          onNextClick={
+            index !== sectionCount - 1
+              ? () => {
+                  trigger(child.props.fields).then((valid) => {
+                    if (valid) {
+                      setActiveIndex(activeIndex + 1);
+                    }
+                  });
+                }
+              : undefined
+          }
+        />
+      );
     }
 
     return child;
   });
+
+  return (
+    <div className={css({ display: 'flex', flexDirection: 'row', height: '100%' })}>
+      <FormSidebar sections={sectionHeaders} onSectionSelect={setActiveIndex} activeIndex={activeIndex} />
+      <div
+        className={css({
+          display: 'flex',
+          flexDirection: 'column',
+          flexGrow: '1',
+          justifyContent: 'space-between',
+        })}
+      >
+        <div className={css({ paddingLeft: '24px' })}>{sections}</div>
```

Comment:
I think we should set the `maxWidth: 800px` here. On my ultrawide the primary action buttons are a mile away from everything else.

![Screenshot of Synthetic Monitoring on a 3440px ultra wide on the check creation page in the new wizard form layout. The primary action buttons are very far to the right at the bottom of the screen. The rest of the content is in the first third of the screen on the left.](https://github.com/grafana/synthetic-monitoring-app/assets/6906380/e512fd73-503e-43d3-9709-78c38b1fc6b8)